mesonbuild/compilers.py

```diff
@@ -2729,3 +2733,135 @@ def unix_compile_flags_to_native(self, args):
 
     def get_link_debugfile_args(self, targetfile):
         return []
+
+
+# Only tested on linux for ICC 17.0.0
+#    * object extraction does not compile (don't know) (25 object extraction fails)
+#    * ICC returns no error if not existing flags are used (i.e. -fsomeflag),
```

Comment:
ah ok, yeah stderr shows a warning, with the failed flag. So a regex could match the corresponding line with warning.

Arguments are tested by `has_argument` and `has_multi_arguments` which calls `compiles`, how can I get the error output?